src/libpriv/rpmostree-core.h

```diff
@@ -82,12 +82,17 @@ namespace rpmostreecxx {
 void core_libdnf_process_global_init();
 }
 
+RpmOstreeContext *rpmostree_context_new_base (OstreeRepo   *repo);
+
 RpmOstreeContext *rpmostree_context_new_client (OstreeRepo   *repo);
 
 RpmOstreeContext *rpmostree_context_new_compose (int basedir_dfd,
                                                  OstreeRepo  *repo,
                                                  rpmostreecxx::Treefile &treefile_rs);
 
+void rpmostree_context_set_cache_root (RpmOstreeContext *self,
```

Comment:
Does this need to be public?

Reply:
Well, where I ended up I guess none if this is going to be necessary if we keep the extensions code using `_new_compose()` which I think we can.

In a previous iteration of https://github.com/coreos/rpm-ostree/pull/2955 the treefile generation was after creating the context, but it doesn't need to be.

So...dunno.  I think this is probably still worth merging but I don't have a strong opinion.